homeassistant/components/tts/__init__.py

```diff
@@ -647,20 +840,24 @@ async def post(self, request: web.Request) -> web.Response:
             data = await request.json()
         except ValueError:
             return self.json_message("Invalid JSON specified", HTTPStatus.BAD_REQUEST)
-        if not data.get(ATTR_PLATFORM) and data.get(ATTR_MESSAGE):
+        if (
+            not data.get("engine_id")
```

Comment:
The `engine_id` parameter is a new optional parameter for the view api that can be used to target the correct engine instead of using `platform`.

If this is ok, the frontend should be updated to use it, if it still needs the view api.